### ISSUE TITLE
Methods defined in steps_file are not added to intellisense and not running

### ISSUE THREAD

**Issue by kirksl** [Codeception/CodeceptJS#3447]
#### What are you trying to achieve?
Expect that if I add a method (eg doSomethingGlobal) to steps_file.js or steps_file.ts
That I can reference it with intellsense in a test (eg I.doSomethingGlobal)
And that I can execute it in a test

#### What do you get instead?
When I add method (eg doSomethingGlobal) to steps_file.js or steps_file.ts
When I try to reference it, it is not found in intellisense
And it does not get executed

Reference project exhibiting the problem
https://github.com/kirksl/codecept-mobile-automation

### Details

* CodeceptJS version: 3.3.6
* NodeJS Version: 16.17.1
* Operating System: Windows 10
* puppeteer || webdriverio || testcafe version (if related): Appium
* Configuration file: See referenced project above

**Comment by kobenguyent** [comment]
hey @kirksl not sure if you already ran `npx codeceptjs def`

**Comment by kirksl** [comment]
@PeterNgTr yes also tried that (attached image)
Same problem after running it (attached image)
Interestingly that command also doesn't add Appium into "WithTranslation"
![image](https://user-images.githubusercontent.com/24948266/195652717-ff96c239-5a5c-446c-b74b-ea748aaa081a.png)
![image](https://user-images.githubusercontent.com/24948266/195653122-4d51eacc-cc8e-4eaf-83c9-d91fa0db23be.png)

**Comment by kobenguyent** [comment]
Oh I remembered that is known issue. I found a workaround here. 

https://github.com/PeterNgTr/ExpectWrapper

**Comment by kirksl** [comment]
Still not seeing method "doSomethingGlobal" added to "I" when trying the workaround.  
![image](https://user-images.githubusercontent.com/24948266/195690277-f0573133-f5d2-40b7-9d65-050f3316be55.png)

**Comment by kobenguyent** [comment]
Hey @kirksl I mean your custom methods should be created and importing like my workaround. 😄

**Comment by kobenguyent** [comment]
Please refer https://github.com/codeceptjs/CodeceptJS/issues/3056

**Comment by kirksl** [comment]
Hey @PeterNgTr thanks for pointing out #3056 

I spent my day completely solving and creating notes for this.
I was able to get everything to work without having to create a package and importing it as well
Hopefully this helps someone!

Notes
1. remove .ts from steps_file import in steps.d.ts or intellisense will break
do this: type steps_file = typeof import('./steps_file');
not this: type steps_file = typeof import('./steps_file.ts');

2. declare main export in steps_file without a return type or intellisense will break
do this: export = function() { }
not this: export = function(): any { }

3. add 'this' statements to new 'I' methods in steps_file or they will not run in codecept execution flow
do this: actor({newMethod: async function() { this.wait(1); }
not this: actor({newMethod: async function() { console.log('abc'); }

My Fixes
https://github.com/kirksl/codecept-mobile-automation/commit/7e6c345a9ffd263ffe43c51a8db516f8e433b13c
